File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 2/27/2026, 5:23:58 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        maxiSum = -inf
        currSum = 0

        for num in nums:
            currSum += num

            if currSum > maxiSum:
                maxiSum = currSum

            if currSum < 0:
                currSum = 0
            

        return maxiSum
        

File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 2/27/2026, 5:24:00 PM
class Solution {
    public int maxSubArray(int[] nums) {
        
        int sum = 0;
        int max = Integer.MIN_VALUE;
        for(int i = 0; i < nums.length; i++){
           sum = sum + nums[i];
           max = Math.max(sum,max);

           if(sum < 0) sum = 0;

        }

        return max;

    }
}